src/qibolab/instruments/rfsoc.py

```diff
@@ -262,12 +365,31 @@ def classify_shots(self, i_values: List[float], q_values: List[float], qubit: Qu
             return [shots]
         return shots
 
+    def base_recursion_layer_sweep(
```

Comment:
Cosmetics: better to name a function after what it does, not what it is used for.